torch/_inductor/codegen/triton.py

```diff
@@ -871,7 +871,21 @@ def index_expr(cls, expr, dtype):
 
     @staticmethod
     def masked(mask, body, other):
-        with V.kernel.mask_loads(mask) as new_mask:
+        nodes = body.graph.find_nodes(op="output")
+
+        need_where = False
+        for node in nodes:
+            for arg in node.args:
+                if arg.target != "load" or V.graph.is_unspec_arg(arg.args[0]):
+                    need_where = True
+
+        if nodes and not need_where:
```

Comment:
can a graph not have an output?

Reply:
I didn't want to assume that each graph does. I'll remove that

Reply:
We already assume the graph has an output when we call it and use the result like a `CSEVariable`.